apps/aecore/src/txs/aec_tx.erl

```diff
@@ -0,0 +1,43 @@
+-module(aec_tx).
+
+-export([apply_signed/3]).
+
+-include("common.hrl").
+-include("trees.hrl").
+-include("txs.hrl").
+
+%% aec_tx behavior callbacks
+
+-callback new(Args :: map(), Trees :: trees()) ->
+    {ok, Tx :: term()}.
+
+-callback run(Tx :: term(), Trees :: trees(), Height :: non_neg_integer()) ->
+    {ok, NewTrees :: trees()}.
+
+
+%% API
+
+-spec apply_signed(list(signed_tx()), trees(), non_neg_integer()) -> {ok, trees()} | {error, term()}.
+apply_signed([], Trees, _Height) ->
+    {ok, Trees};
+apply_signed([SignedTx | Rest], Trees0, Height) ->
+    case aec_tx_sign:verify(SignedTx) of
+        ok ->
+            Tx = aec_tx_sign:data(SignedTx),
+            case aec_proofs:prove(Tx, Trees0) of
```

Comment:
I understand this line can be deleted.

Reply:
@lucafavatella this comment is regarding our decision to abandon proofs?

Reply:
Well, we still need to check whether an account has enough funds, etc? This is what I meant to have done here.

Reply:
> @lucafavatella this comment is regarding our decision to abandon proofs?

@sennui  Not really. I wrote the comment as I was not seeing the intent of this line, that @zp clarified in the following comment...

> Well, we still need to check whether an account has enough funds, etc? This is what I meant to have done here.

@zp I see. So if I understand you correctly a more explicit name would be `verify_tx_assumptions` or `verify_tx_preconditions`, instead of `aec_proofs:prove` (the point is verifying the preconditions, and I understand we do that on the state trees - not proofs). So each transaction module, in addition to the APIs `new` and `run`, would have an API `verify_preconditions`.

Reply:
I understand that this can be done from within a transaction (in `run` function). Let's see later if we need additional tx behaviour function for it.